src/SFA.DAS.RoatpGateway.Domain/PaginatedList.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace SFA.DAS.RoatpGateway.Domain
+{
+    public abstract class PaginatedList
+    {
+        public abstract int PageIndex { get; }
+        public abstract int PageSize { get; }
+        public abstract int PageSetSize { get; set; }
+
+        public abstract int TotalRecordCount { get; }
+        public abstract int TotalPages { get; }
+
+        public abstract bool HasPreviousPage { get; }
+        public abstract bool HasNextPage { get; }
+
+        public abstract int FirstVisiblePage { get; }
+        public abstract int LastVisiblePage { get; }
+
+        public abstract int FirstVisibleItem { get; }
+        public abstract int LastVisibleItem { get; }
```

Comment:
Some of these properties are abstract but their deriving class doesn't provide any special implementation. Wouldn't it be better to have them as virtual ?

Reply:
That was lifted and shifted from Admin Service, no idea really